File: Is Subsequence - Leetcode 392/Is Subsequence - Leetcode 392.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int sp = 0;
        int tp = 0;

        while (sp < s.length() && tp < t.length()) {
            if (s[sp] == t[tp]) {
                sp++;
            }
            tp++;
        }

        return sp == s.length();        
    }
};

File: Is Subsequence - Leetcode 392/Is Subsequence - Leetcode 392.go
package solution

func isSubsequence(s string, t string) bool {
    i := 0
    j := 0
    for i<len(s) && j < len(t){
        if s[i] == t[j]{
            i++
        }
        j++
    }
    return len(s)==i
}
